src/cmap/auth/mongo_credentials.ts

```diff
@@ -137,6 +142,44 @@ export class MongoCredentials {
       throw new MongoMissingCredentialsError(`Username required for mechanism '${this.mechanism}'`);
     }
 
+    if (this.mechanism === AuthMechanism.MONGODB_OIDC) {
+      if (this.username) {
+        throw new MongoAPIError(
```

Comment:
I think its valid to make these all `MongoInvalidArgumentError` slightly more specific about a misconfiguration issue.

Reply:
Changed all to `MongoInvalidArgumentError`.